test/test_iteratees.ml

```diff
@@ -0,0 +1,127 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open OUnit
+open Wslib
+
+module I = Iteratees.Iteratee(Test.StringMonad)
+module TestWsIteratee = Websockets.Wsprotocol(Test.StringMonad)
+open TestWsIteratee
+open I
+
+let teststr = "\x81\x05\x48\x65\x6c\x6c\x6f"
+let teststr1 = teststr ^ "\x88\x00"
+let it = wsunframe (writer Test.StringMonad.strwr "foo")
+
+let assert_ready_state state () = assert_equal state "Ready"
+let assert_done_state state () = assert_equal state "Done"
```

Comment:
I think these tests have a problem similar to the one for the websockets. Here instead of checking the state of the iteratee, we should be checking that the various functions do what they claim. You could use the `enum_` functions for this, e.g.

```ocaml
# enum_1chunk "test" (heads "t");;
- : int t StringMonad.t = {StringMonad.data = IE_done 1; str = ""}                                                                                                                       

# enum_1chunk "test" (heads "t");;
- : int t StringMonad.t = {StringMonad.data = IE_done 1; str = ""}                                                                                                                       

# enum_1chunk "test" (heads "te");;
- : int t StringMonad.t = {StringMonad.data = IE_done 2; str = ""}                                                                                                                       

# enum_1chunk "test" (heads "g");;
- : int t StringMonad.t = {StringMonad.data = IE_done 0; str = ""}
```
 And you can use the `StringMonad.get_data` anf `get_str` to check that the correct thing happened.
@jonludlam do you have a better recommendation?

Reply:
@mseri thanks for the review, I have updated and kept few of iteratee tests. Could you please help to review and put a closure on this PR.